bytecomp/translmod.ml

```diff
@@ -342,9 +353,82 @@ let transl_class_bindings cl_list =
        (id, transl_class ids id meths cl vf))
      cl_list)
 
+(* Compile one or more functors, merging curried functors to produce
+   multi-argument functors.  Any [@inline] attribute on a functor that is
+   merged must be consistent with any other [@inline] attribute(s) on the
+   functor(s) being merged with.  Such an attribute will be placed on the
+   resulting merged functor. *)
+
+let merge_inline_attributes attr1 attr2 loc =
+  match Lambda.merge_inline_attributes attr1 attr2 with
+  | Some attr -> attr
+  | None -> raise (Error (loc, Conflicting_inline_attributes))
+
+let merge_functors mexp coercion root_path =
+  let rec merge mexp coercion path acc inline_attribute =
+    let finished () = acc, mexp, path, coercion, inline_attribute in
+    match mexp.mod_type with
+    | Mty_alias _ -> finished ()
```

Comment:
This wasn't needed because anything that is a `Tmod_functor` isn't an alias.

However we did spot another potential bug in `transl_module` which has now been fixed by this patch.  A "present" alias could cause generation of `lambda_unit` for the module expression which is wrong.  This probably can't happen at the moment but under @lpw25 's proposed `module type of` changes could do.  We've restricted the match for generating `lambda_unit` to only absent aliases.